.github/workflows/tests.yml

```
@@ -0,0 +1,27 @@
+name: .NET Tests
+
+on:
+  push:
+    branches: [ "main" ]
```

У вас тест не запустился на выполнения. Лучше добавить еще событие открытия PR.
```suggestion
on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]
```